cmd/common/utils/version-checker.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"github.com/blang/semver"
+	log "github.com/sirupsen/logrus"
+)
+
+type VersionCheckParams struct {
+	ClientVersion         semver.Version
+	ServerVersion         string
+	SemanticVersionErrMsg string
+	VersionSkewWarningMsg string
+}
```

Comment:
I don't think we need to pass a lot of parameters to this function. What about something like:

```diff
diff --git cmd/common/utils/version-checker.go cmd/common/utils/version-checker.go
index e20329c85..4e29045f6 100644
--- cmd/common/utils/version-checker.go
+++ cmd/common/utils/version-checker.go
@@ -1,30 +1,24 @@
 package utils
 
 import (
+	"fmt"
+
 	"github.com/blang/semver"
-	log "github.com/sirupsen/logrus"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/version"
 )
 
-type VersionCheckParams struct {
-	ClientVersion         semver.Version
-	ServerVersion         string
-	SemanticVersionErrMsg string
-	VersionSkewWarningMsg string
-}
+func CheckServerVersionSkew(serverVersion string) error {
+	clientSemver := version.Version()
 
-func CheckVersionSkew(vc VersionCheckParams) {
-	if len(vc.ClientVersion.Pre) > 0 {
-		return
-	} else {
-		serverSemver, err := semver.Make(vc.ServerVersion)
-		if err != nil {
-			log.Errorf(vc.SemanticVersionErrMsg, err)
-		}
-		if !vc.ClientVersion.Equals(serverSemver) {
-			log.Warnf(
-				vc.VersionSkewWarningMsg,
-				serverSemver, vc.ClientVersion,
-			)
-		}
+	// Do not print any warning if this is a prerelease to avoid annoying developers.
+	if len(clientSemver.Pre) > 0 {
+		return nil
 	}
+
+	serverSemver, err := semver.Make(serverVersion)
+	if err != nil {
+		return fmt.Errorf("parsing semver: %w", err)
+	}
+
+	return fmt.Errorf("version skew detected: client (%s) vs server (%s)", clientSemver, serverSemver)
 }
diff --git cmd/gadgetctl/main.go cmd/gadgetctl/main.go
index 69e4cd9f8..581aaff6c 100644
--- cmd/gadgetctl/main.go
+++ cmd/gadgetctl/main.go
@@ -26,7 +26,6 @@ import (
 	"github.com/inspektor-gadget/inspektor-gadget/cmd/common"
 	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
 	"github.com/inspektor-gadget/inspektor-gadget/internal/deployinfo"
-	"github.com/inspektor-gadget/inspektor-gadget/internal/version"
 	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/all-gadgets"
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/environment"
 	grpcruntime "github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/grpc"
@@ -81,14 +80,9 @@ func main() {
 		info, err := deployinfo.Load()
 		if err != nil {
 			log.Warnf("Failed to load deploy info: %s", err)
+		} else if err := commonutils.CheckServerVersionSkew(info.ServerVersion); err != nil {
+			log.Warnf(err.Error())
 		}
-		checkParams := commonutils.VersionCheckParams{
-			ClientVersion:         version.Version(),
-			ServerVersion:         info.ServerVersion,
-			SemanticVersionErrMsg: "Failed to create a semantic version object for the gadget daemon version: %s",
-			VersionSkewWarningMsg: "Version skew detected (gadget daemon v%s vs gadgetctl v%s)",
-		}
-		commonutils.CheckVersionSkew(checkParams)
 	}
 
 	hiddenColumnTags := []string{"kubernetes"}
diff --git cmd/kubectl-gadget/main.go cmd/kubectl-gadget/main.go
index 2f4406687..865272098 100644
--- cmd/kubectl-gadget/main.go
+++ cmd/kubectl-gadget/main.go
@@ -23,7 +23,6 @@ import (
 
 	// Import this early to set the enrivonment variable before any other package is imported
 	"github.com/inspektor-gadget/inspektor-gadget/internal/deployinfo"
-	"github.com/inspektor-gadget/inspektor-gadget/internal/version"
 	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/environment/k8s"
 	paramsPkg "github.com/inspektor-gadget/inspektor-gadget/pkg/params"
 
@@ -131,14 +130,9 @@ func main() {
 		info, err := deployinfo.Load()
 		if err != nil {
 			log.Warnf("Failed to load deploy info: %s", err)
+		} else if err := commonutils.CheckServerVersionSkew(info.ServerVersion); err != nil {
+			log.Warnf(err.Error())
 		}
-		checkParams := commonutils.VersionCheckParams{
-			ClientVersion:         version.Version(),
-			ServerVersion:         info.ServerVersion,
-			SemanticVersionErrMsg: "Failed to create a semantic version object for the gadget pods version: %s",
-			VersionSkewWarningMsg: "Version skew detected (gadget pods v%s vs kubectl-gadget v%s), use 'kubectl gadget deploy' to fix it.\n",
-		}
-		commonutils.CheckVersionSkew(checkParams)
 	}
 
 	namespace, _ := utils.GetNamespace()

```